lib/src/validationEngine.ts

```diff
@@ -71,61 +73,59 @@ export class ValidationEngine implements IValidationEngine {
         })
         .catch((result) => {
           // Build failed validation Result
-          var errorInformation = `Uncontrolled error when validating full form, check custom validations code`;
+          const errorInformation = `Uncontrolled error when validating full form, check custom validations code`;
           console.log(errorInformation);
-          reject(errorInformation);
+          reject(result);
         });
-
     });
 
     return fullFormValidatedPromise;
   }
 
-  triggerFieldValidation(vm: any, key: string, value: any, filter: any = consts.defaultFilter): Promise<FieldValidationResult> {
+  triggerFieldValidation(vm: any, key: string, value: any, filter: ValidationFilters = consts.defaultFilter): Promise<FieldValidationResult> {
```

Comment:
Ok, sorry but I mean, you have declared triggerFieldValidation as `triggerFieldValidation(vm: any, key: string, value: any, filter?: ValidationFilters, customParams?: Object)`. So we have to remove from declaration